aptos-move/framework/aptos-framework/sources/Signature.move

```diff
@@ -21,4 +21,43 @@ module AptosFramework::Signature {
         public_key: vector<u8>,
         message: vector<u8>
     ): bool;
+
+    /// Recovers the signer public key from a secp256k1 `signature` provided the `recovery_id` and signed
+    /// `message` (32 byte digest).
+    /// Returns `(public_key, true)` if inputs are valid and `([], false)` if invalid.
+    native public fun secp256k1_recover(
+        message: vector<u8>,
+        recovery_id: u8,
+        signature: vector<u8>
+    ): (vector<u8>, bool);
+
+    #[test]
+    /// Test on a valid signature created using pk = x"0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef"
+    fun test_secp256k1() {
```

Comment:
Can we add a randomized test here: 

- pick a random key-pair, 
- pick and sign a random message, 
- call recover
- check it returns the PK.

Reply:
...otherwise, it looks good.